Cython/Build/Tests/TestRecythonize.py

```diff
@@ -1,212 +1,218 @@
-import shutil
 import os
+import shutil
 import tempfile
 import time
 
 import Cython.Build.Dependencies
 import Cython.Utils
 from Cython.TestUtils import CythonTest
 
+SAME = "The result of cytonization is the same"
+INCORRECT = "Incorrect cythonization"
+LINE_BEFORE_IMPLEMENTATION = '  /* "{filename}":{at_line}\n'
+VARS_LINE = '  /*--- Wrapped vars code ---*/\n'
 
-def fresh_cythonize(*args, **kwargs):
-    Cython.Utils.clear_function_caches()
-    Cython.Build.Dependencies._dep_tree = None  # discard method caches
-    Cython.Build.Dependencies.cythonize(*args, **kwargs)
 
 class TestRecythonize(CythonTest):
+    language_level = 3
+    dep_tree = Cython.Build.Dependencies.create_dependency_tree()
+
+    def clear_function_and_Dependencies_caches(self):
+        Cython.Utils.clear_function_caches()
+        Cython.Build.Dependencies._dep_tree = None  # discard method caches
 
     def setUp(self):
         CythonTest.setUp(self)
+        self.clear_function_and_Dependencies_caches()
         self.temp_dir = (
             tempfile.mkdtemp(
                 prefix='recythonize-test',
                 dir='TEST_TMP' if os.path.isdir('TEST_TMP') else None
             )
         )
+        self.src_dir = os.path.join(self.temp_dir, 'src')
+        os.mkdir(self.src_dir)  # self.temp_dir is already a temp
```

Comment:
It took me a while to understand what you want to say with this comment. It's clearer without it, I think.
```suggestion
        os.mkdir(self.src_dir)
```